bin/proving-service/src/main.rs

```diff
@@ -46,14 +46,19 @@ mod test {
 
     use crate::{
         api::ProverRpcApi,
-        generated::{api_client::ApiClient, api_server::ApiServer, ProveTransactionRequest},
+        commands::worker::ProverTypeSupport,
+        generated::{api_client::ApiClient, api_server::ApiServer, ProofType, ProvingRequest},
     };
 
     #[tokio::test(flavor = "multi_thread", worker_threads = 3)]
     async fn test_prove_transaction() {
         // Start the server in the background
         let listener = TcpListener::bind("127.0.0.1:50052").await.unwrap();
-        let api_service = ApiServer::new(ProverRpcApi::default());
+
+        let mut prover_type = ProverTypeSupport::default();
+        prover_type.with_transaction();
```

Comment:
If we make `with_transaction` take `mut self` instad and return `Self`, then we could write this in one line. Do you think there are any downsides to that?

Reply:
No, I do not see any downside 👌🏼